src/util/triplea/MapXMLCreator/ImageScrollPanePanel.java

```diff
@@ -0,0 +1,309 @@
+
+			@Override
+			public void mouseClicked(final MouseEvent e)
+			{
+				s_instance.mouseClickedOnImage(centers, m_imagePanel, e);
+			}
+			
+			public void mouseDragged(final MouseEvent e)
+			{
+				final JViewport vport = (JViewport) m_imagePanel.getParent();
+				final Point cp = e.getPoint();
+				final Point vp = vport.getViewPosition();
+				vp.translate(pp.x - cp.x, pp.y - cp.y);
+				SwingUtilities.invokeLater(new Runnable()
+				{
+					
+					@Override
+					public void run()
+					{
+						
+						m_imagePanel.scrollRectToVisible(new Rectangle(vp, vport.getSize()));
+					}
+				});
+				pp.setLocation(cp);
+			}
+			
+			public void mousePressed(MouseEvent e)
+			{
+				m_imagePanel.setCursor(hndCursor);
+				pp.setLocation(e.getPoint());
+			}
+			
+			public void mouseReleased(MouseEvent e)
+			{
+				m_imagePanel.setCursor(defCursor);
+				pp.setLocation(e.getPoint());
+			}
+		};
+		m_imagePanel.addMouseListener(imageMouseAdapter);
+		m_imagePanel.addMouseMotionListener(imageMouseAdapter);
+		
+		return m_imagePanel;
+	}
+	
+	private static void loadPolygons()
+	{
+		if (s_polygonsInvalid)
+		{
+			s_polygons.clear();
+			s_polygonsInvalid = false;
+		}
+		if (!s_polygons.isEmpty())
+			return;
+		File file = null;
+		if (MapXMLCreator.s_mapPolygonsFile == null)
+		{
+			if (MapXMLCreator.s_mapFolderLocation != null && MapXMLCreator.s_mapFolderLocation.exists())
+				file = new File(MapXMLCreator.s_mapFolderLocation, "polygons.txt");
+			if (file == null || !file.exists())
+				file = new File(MapXMLCreator.s_mapImageFile.getParent() + File.separator + "polygons.txt");
+		}
+		else
+		{
+			file = MapXMLCreator.s_mapPolygonsFile;
+		}
+		if (MapXMLCreator.s_mapPolygonsFile != null
+					|| file.exists()
+					&& JOptionPane.showConfirmDialog(new JPanel(), "A polygons.txt file was found in the map's folder, do you want to use the file to supply the territory shapes?", "File Suggestion",
+								1) == 0)
+		{
+			try
+			{
+				System.out.println("Load Polygons from " + file.getPath());
+				s_polygons = PointFileReaderWriter.readOneToManyPolygons(new FileInputStream(file.getPath()));
+			} catch (final IOException ex1)
+			{
+				System.out.println("Something wrong with your Polygons file");
+				ex1.printStackTrace();
+			}
+		}
+		else
+		{
+			try
+			{
+				System.out.println("Select the Polygons file");
+				final String polyPath = new FileOpen("Select A Polygon File", MapXMLCreator.s_mapFolderLocation, ".txt").getPathString();
+				if (polyPath != null)
+				{
+					System.out.println("Polygons : " + polyPath);
+					s_polygons = PointFileReaderWriter.readOneToManyPolygons(new FileInputStream(polyPath));
+				}
+				else
+				{
+					System.out.println("Polygons file not given. Will run regardless");
+				}
+			} catch (final IOException ex1)
+			{
+				System.out.println("Something wrong with your Polygons file");
+				ex1.printStackTrace();
+			}
+		}
+		MapXMLCreator.s_mapPolygonsFile = file;

```

Comment:
This method is using side-effects. Rather than doing this assignment in the method, have it return the file.
Then the caller woudl look liek this:
`MapXMLCreator.s_mapPolygonsFile = loadPolygons()`
Now it is clear what is happening, rather than
`loadPolygons()`, which a reader then has to go through to understand. The method may as well be inlined at that point.